tests/unit/multi_select_test.coffee

```diff
@@ -0,0 +1,129 @@
+    ok isHidden(find '.ember-select-results', multiSelectComponent,
+      messageHidden)
+
+  validateTextFieldFocus = (messageFocus) ->
+    ok isFocused(textField,multiSelectComponent, messageFocus)
+
+  multiSelectComponent.focus()
+  wait().then ->
+    # test pressing ENTER key to open dropdown
+    pressEnter(multiSelectComponent)
+  .then ->
+    validateDropdownVisible('Dropdown list should appear after pressing
+      Enter')
+    resultItems = find '.ember-select-result-item', multiSelectComponent
+    ok($(resultItems[0]).hasClass('highlighted'),
+      'The first option should be highlighted')
+    # test selecting option using ENTER key
+    selectedText = $(resultItems[0]).text().trim()
+
+    pressEnter(multiSelectComponent)
+  .then ->
+    validateTextFieldFocus('Text field should still be focused after
+      selecting using Enter')
+    # test if selected Item is actually selected
+    resultItems = find '.ember-select-search-choice', multiSelectComponent
+    currentText = $(resultItems[resultItems.length - 1]).text().trim()
+    # we are using character "x" to make it look like a close button so
+    # here we need to trim this letter at the end of the select choice text
+    equal(currentText.substring(0, currentText.length - 2), selectedText,
+      'The current highlighted option should be the last choice pill in
+      the list')
+
+    # open dropdown when clicking on the text field
+    click(textField)
+  .then ->
+    validateDropdownVisible('Dropdown list should appear after clicking
+      on the input field')
+
+    # make one more selection
+    pressEnter(multiSelectComponent)
+  .then ->
+    resultItems = find '.ember-select-search-choice', multiSelectComponent
+    equal(resultItems.length, 2, 'There should be 2 selected items')
+
+    # test deleting using keyboard
+    pressBackspace(textField)
+  .then ->
+    resultItems = find '.ember-select-search-choice', multiSelectComponent
+    equal(resultItems.length, 1,
+      'There should be 1 selected item after deleting one')
+    validateTextFieldFocus('Dropdown list should still be shown after
+      deleting using keyboard','Text field should be focused after
+      pressing delete')
+
+    # test pressing Spacebar to open the dropdown
+    pressSpacebar(multiSelectComponent)
+  .then ->
+    validateDropdownVisible('Dropdown list should appear after pressing
+      Spacebar')
+
+    # test adding using mouse
+    resultItems = find '.ember-select-result-item', multiSelectComponent
+    click(resultItems[0])
+  .then ->
+    resultItems = find '.ember-select-search-choice', multiSelectComponent
+    equal(resultItems.length, 2, 'There should be 2 selected items after
+      selecting one item using mouse')
+    validateTextFieldFocus('Text field should be focused after selecting
+      using mouse')
+
+    # test deleting one selected item using mouse
+    closeButtons = find '.ember-select-search-choice-close', multiSelectComponent
```

Comment:
@cyril-sf which item will it click on assuming that `find '.ember-select-search-choice-close', multiSelectComponent` returns multiple results. I guess it will click on the 1st one?